runtime/vm/createramclass.cpp

```diff
@@ -3451,10 +3451,9 @@ internalCreateRAMClassFromROMClassImpl(J9VMThread *vmThread, J9ClassLoader *clas
 					 * elements may be flattened arrays.
 					 */
 					U_32 arrayFlags = J9ClassLargestAlignmentConstraintReference | J9ClassLargestAlignmentConstraintDouble;
-					if (J9_ARE_ALL_BITS_SET(javaVM->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_ENABLE_VT_ARRAY_FLATTENING)) {
-						/* TODO restrict this flag to be set only for null-restricted
-						 * arrays once value type command line tests are updated.
-						 */
+					if (J9_ARE_ALL_BITS_SET(options, J9_FINDCLASS_FLAG_CLASS_OPTION_NULL_RESTRICTED_ARRAY)
```

Comment:
`J9_FINDCLASS_FLAG_CLASS_OPTION_NULL_RESTRICTED_ARRAY` is been checked here and line 3460. The code can be optimized to check it only once.

Reply:
That is intentional otherwise J9ClassArrayIsNullRestricted will not make it past
```
ramArrayClass->classFlags |= (elementClass->classFlags & arrayFlags);
```

Reply:
I am thinking about something like:

```
if (J9_ARE_ALL_BITS_SET(options, J9_FINDCLASS_FLAG_CLASS_OPTION_NULL_RESTRICTED_ARRAY) {
   if (J9_ARE_ALL_BITS_SET(javaVM->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_ENABLE_VT_ARRAY_FLATTENING) {
        arrayFlags |= J9ClassIsFlattened;
    }
    ramArrayClass->classFlags |= J9ClassArrayIsNullRestricted;
}
ramArrayClass->classFlags |= (elementClass->classFlags & arrayFlags);
```

Reply:
Okay that works.